### PR TITLE
[BF-112] Create global error handling

### PR DESCRIPTION
**Pull request by alexkocharyan01**

### CODE REVIEW

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  useEffect(() => {
+    if (error) {
+      dispatch(
+        viewsMiddleware.openModal({
+          name: ModalName.getErrorMessage,
+          props: {
+            error,
+          },
+        })
```
Comment (ArthurO2002) [inline src/components/ErrorMessagePopup/ErrorMessagePopup.tsx:17]: We should open toastNotificationPopUp from redux

File: src/components/Modals/Error/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              fill="none"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
+                d="M6 18L18 6M6 6l12 12"
```
Comment (ArthurO2002) [inline src/components/Modals/Error/index.tsx:46]: Put svg in another file

File: src/redux/slices/views/reducers.ts [hunk 3]
```diff
@@ -32,6 +32,12 @@ const reducers = createReducer({
   updateToastNotificationState<P>(state: ViewsProps, action: IAction<IOpenedAlert<P>>) {
     state.toastNotificationPopUp = action.payload ? action.payload : { open: false, props: {} }
   },
+  setError(state: ViewsProps, action: IAction<string>) {
+    state.error.error = action.payload
+  },
+  resetError(state: ViewsProps, action: IAction<null>) {
+    state.error.error = action.payload
+  },
 })
```
Comment (ArthurO2002) [inline src/redux/slices/views/reducers.ts:41]: Not sure that we need this

File: src/redux/slices/views/middleware.ts [hunk 4]
```diff
@@ -46,6 +48,14 @@ const setToastNotificationPopUpState =
...
   }
 
+const setErrorMessage = (value: string) => (dispatch: AppDispatch) => {
+  dispatch(setError(value))
+}
+
+const resetErrorMessage = () => (dispatch: AppDispatch) => {
+  dispatch(resetError(null))
+}
+
```
Comment (ArthurO2002) [inline src/redux/slices/views/middleware.ts:58]:
```suggestion
const resetErrorMessage = () => (dispatch: AppDispatch) => {
  dispatch(setError(null))
}

```

File: tailwind.config.js [hunk 5]
```diff
@@ -39,7 +39,11 @@ module.exports = {
           light: '#666',
           DEFAULT: '#000',
         },
-        red: '#FF6868',
+        red: {
+          DEFAULT: '#FF6868',
+          light: '#FEE2E2',
+          thin: '#991B1B',
+        },
```
Comment (ArthurO2002) [inline tailwind.config.js:46]:
```suggestion
        error: {
          DEFAULT: '#FF6868',
          light: '#FEE2E2',
          thin: '#991B1B',
        },
```

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      )
+    }
+  }, [error])
+
+  return null
```
Comment (ArthurO2002) [inline src/components/ErrorMessagePopup/ErrorMessagePopup.tsx:22]: Here should be your Popup

File: src/manager/axiosInstance.ts [hunk 7]
```diff
@@ -64,6 +64,8 @@ class RequestManager {
               viewsMiddleware.setRedirectionState({ path: '/login', params: '', apply: true })
             )
           }
+        } else {
+          dispatch(viewsMiddleware.setErrorMessage(error.response.data.status.message))
```
Comment (ArthurO2002) [inline src/manager/axiosInstance.ts:68]: Here should be opened the popup

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  const error = useAppSelector(viewsSelector.error)
+
+  useEffect(() => {
+    if (error) {
+      dispatch(
+        viewsMiddleware.openModal({
+          name: ModalName.getErrorMessage,
+          props: {
+            error,
+          },
```
Comment (ArthurO2002) [inline src/components/ErrorMessagePopup/ErrorMessagePopup.tsx:16]: Move this logic to axiosInstance

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]
We also need to remove all previous error messages that we had

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx (hunk 8 above)
Comment (alexkocharyan01) [reply src/components/ErrorMessagePopup/ErrorMessagePopup.tsx]: done

File: src/manager/axiosInstance.ts (hunk 7 above)
Comment (alexkocharyan01) [reply src/manager/axiosInstance.ts]: done

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx (hunk 6 above)
Comment (alexkocharyan01) [reply src/components/ErrorMessagePopup/ErrorMessagePopup.tsx]: i remove it

File: tailwind.config.js (hunk 5 above)
Comment (alexkocharyan01) [reply tailwind.config.js:46]: we use red color in a another component

File: src/redux/slices/views/middleware.ts (hunk 4 above)
Comment (alexkocharyan01) [reply src/redux/slices/views/middleware.ts]: i remove it

File: src/redux/slices/views/reducers.ts (hunk 3 above)
Comment (alexkocharyan01) [reply src/redux/slices/views/reducers.ts]: i remove it

File: src/components/Modals/Error/index.tsx (hunk 2 above)
Comment (alexkocharyan01) [reply src/components/Modals/Error/index.tsx]: done

File: src/components/ErrorMessagePopup/ErrorMessagePopup.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/components/ErrorMessagePopup/ErrorMessagePopup.tsx]: done

**Pull request merged**